src/types/mod.rs

```diff
@@ -411,6 +411,58 @@ impl Display for QualifiedUrl {
     }
 }
 
+derive_or_from_str! {
+    /// A publisher, possibly with a location.
+    #[derive(Clone, Debug, PartialEq, Eq, Hash)]
+    pub struct Publisher where "FormatString string or dictionary with \"name\" and \"location\"" {
+        /// Publisher of the item.
+        name: Option<FormatString>,
+        /// Physical location at which the item was published or created.
+        location: Option<FormatString>,
+    }
+}
+
+impl Serialize for Publisher {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        if let Some(location) = &self.location {
+            let mut map = serializer.serialize_map(Some(2))?;
+            map.serialize_entry("name", &self.name)?;
+            map.serialize_entry("location", location)?;
+            map.end()
+        } else {
+            self.name.serialize(serializer)
+        }
+    }
+}
+
+impl Publisher {
+    /// Create a new publisher.
+    pub fn new(name: Option<FormatString>, location: Option<FormatString>) -> Self {
+        Self { name, location }
+    }
+
+    /// Publisher of the item.
+    pub fn name(&self) -> Option<&FormatString> {
+        self.name.as_ref()
+    }
+
+    /// Physical location at which the item was published or created.
+    pub fn location(&self) -> Option<&FormatString> {
+        self.location.as_ref()
+    }
+}
+
+impl FromStr for Publisher {
+    type Err = ChunkedStrParseError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
```

Comment:
Could consider adding a short doc comment here explaining that the parsed string will correspond to the publisher's name.